device/iot-device-samples/multiplexing-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/ClientManagerBase.java

```diff
@@ -63,7 +64,7 @@ public void close()
 
     public void closeNow()
     {
-        synchronized (lastKnownConnectionStatus)
```

Comment:
Not generally a good idea to synchronize on non-final objects. Since this state is supposed to change over time, it makes more sense to add a lock to synchronize on instead